app/models/case_cycle.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# The cycle for subsidy cases
+class CaseCycle < UuidApplicationRecord
+  monetize :copay_cents
+
+  enum copay_frequency: { daily: 'daily', weekly: 'weekly', monthly: 'monthly' }
+  enum status: { submitted: 'submitted', pending: 'pending', approved: 'approved', denied: 'denied' }
+
+  before_save :set_slug
+
+  validates :slug, uniqueness: true
+  validates :copay, numericality: { greater_than: 0 }
+  validates :submitted_on, date_param: true, allow_blank: true
```

Comment:
add `validates :copay_frequency, inclusion: { in: copay_frequency.keys }`
and `validates :status, inclusion: { in: status.keys }`

Reply:
When it is present, the spec `it { should validate_inclusion_of(:status).in_array(CaseCycle::STATUSES) }` fails because when you try to initialize an object with an invalid status (`'test'`, for instance), ActiveRecord enum throws an `ArgumentError ('test' is not a valid status)`. Check https://github.com/rails/rails/issues/13971 for more information.

Reply:
<sigh>  Wish they (rails) would re-open that issue and make sure it's addressed.